vtex/faststore
# Add showSponsored as api configuration on faststore.config.js to IntelligentSearch integration

## What's the purpose of this pull request?

This PR adds support for showing sponsored products when using Intelligent Search. FastShop and probably more retailers are interested in our ad solution (VTEX Ad Network). This PR intends to create a frictionless integration between FastStore and VTEX Ad Network.

## How it works?

The store will change the showSponsored configuration on the feaststore.config.js file to allow intelligent search to return sponsored products. This will only happen if the store is a active publisher on VTEX Ad Network.

On faststore.config.js
```yaml
 // Platform specific configs for API
  API: {
    ...
    showSponsored: true,
  }
```

## How to test it?

* Change the `showSponsored` attribute to `true` on faststore.config.js
* Verify if the store is an active publisher on VTEX Ad Network
* Create a campaign to be delivered to the store with a valid product

Check the starter page with the `showSponsored` flag to true and no label: [Starter With Ads](https://starter-5q5m6v7l7-faststore.vercel.app/office) - related [PR](https://github.com/vtex-sites/starter.store/pull/540):

Example with no label:
![image](https://github.com/user-attachments/assets/119baa57-a5b8-4b7f-99c1-954aaae30e07)

Perfomance for category page:
![image](https://github.com/user-attachments/assets/714db14c-a3b4-435c-9acd-5c34f43d37b6)
![image](https://github.com/user-attachments/assets/e9d64b9d-aba5-4987-8477-21ebd6f0ce08)

Performance for full text search page:
![image](https://github.com/user-attachments/assets/de06dc1a-c2a5-47ec-814b-5ef6dada863c)
![image](https://github.com/user-attachments/assets/ce2c910f-6b46-48d2-91a6-fbe6fd62b0d6)


### Starters Deploy Preview


## References

* [VTEX Ad Network](https://developers.vtex.com/docs/api-reference/vtex-ad-network-api#get-/sponsored_products/-facets-)
* [Intelligent Search API](https://developers.vtex.com/docs/api-reference/intelligent-search-api#get-/product_search/-facets-)
